### PR TITLE
Add integration tests for label verification

### PR DESCRIPTION
**Pull request by NautiluX**
jira: [OSD-10050](https://issues.redhat.com/browse/OSD-10050)

### CODE REVIEW

File: int/integration.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+			panic(err)
+		}
+	}()
+
+	// Wait for cache to start. Is there a better way?
+	time.Sleep(2 * time.Second)
+	return &i, nil
+}
+
+func (i *Integration) Shutdown() {
```
Comment (shibumi) [inline int/integration.go:53]: Is this supposed to be empty?

File: int/int_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	i.Shutdown()
+})
+var _ = Describe("Integrationtests", func() {
+
+	Context("When adding a label to a MachineSet", func() {
+		var (
+			TestLabel string
+			TestValue string
+		)
+		Context("When the labels doesn't exist on the Node", func() {
```
Comment (shibumi) [inline int/int_test.go:34]: Small nit: "When the labels don't exist on the Node" or "When the label doesn't exist on the Node"

File: int/integration.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	// Filter out irrelevant machines (deleting/mismatch labels)
+	machines := []*machinev1.Machine{}
+	for idx := range allMachines.Items {
+		machine := &allMachines.Items[idx]
+		if shouldExcludeMachine(machineSet, machine) {
+			continue
+		}
+
```
Comment (shibumi) [inline int/integration.go:96]: I would remove the empty lines, but that's personal preference, I guess :D

File: int/integration.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	if !selector.Matches(labels.Set(machine.Labels)) {
+		return false
+	}
+
+	return true
+}
+
+func (i *Integration) GetNodeForMachine(m *machinev1.Machine) (v1.Node, error) {
```
Comment (shibumi) [inline int/integration.go:139]: You are exporting the function, but there is no docstring describing it

File: int/integration.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	}
+	node := &v1.Node{}
+	err := i.Client.Get(context.TODO(), types.NamespacedName{Name: m.Status.NodeRef.Name}, node)
+	if err != nil {
+		return v1.Node{}, err
+	}
+	return *node, err
+}
+
+func (i *Integration) DisableWebhook() error {
```
Comment (shibumi) [inline int/integration.go:151]: Missing docstring

File: int/integration.go [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+	for _, ms := range msList.Items {
+		role, ok := ms.Labels["hive.openshift.io/machine-pool"]
+		if ok && role == "worker" {
+			return ms, nil
+		}
+	}
+	return machinev1.MachineSet{}, fmt.Errorf("no worker MachineSet found")
+}
+
+func (i *Integration) GetMachinesForMachineSets(machineSet *machinev1.MachineSet) ([]*machinev1.Machine, error) {
```
Comment (shibumi) [inline int/integration.go:71]: Missing docstring

File: int/integration.go [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+	// Wait for cache to start. Is there a better way?
+	time.Sleep(2 * time.Second)
+	return &i, nil
+}
+
+func (i *Integration) Shutdown() {
+}
+
+func (i *Integration) GetWorkerMachineSet() (machinev1.MachineSet, error) {
```
Comment (shibumi) [inline int/integration.go:56]: Missing docstring

File: int/integration.go (hunk 1 above)
Comment (shibumi) [reply int/integration.go:53]: Missing docstring

File: int/integration.go [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type Integration struct {
+	Client client.Client
+	mgr    manager.Manager
+}
+
+func NewIntegration() (*Integration, error) {
```
Comment (shibumi) [inline int/integration.go:27]: Missing docstring

File: int/integration.go [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type Integration struct {
```
Comment (shibumi) [inline int/integration.go:22]: Some docstrings would come handy.. or is this functionality that does not need to get documented? :)

File: int/int_test.go [hunk 10]
```diff
@@ -0,0 +1,65 @@
+package int_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	//nolint:typecheck // this import is used for general prettines, and currently kept
```
Comment (shibumi) [inline int/int_test.go:10]: s/prettines/prettiness/

File: int/integration.go (hunk 3 above)
Comment (NautiluX) [reply int/integration.go]: That's copied/moved code, don't want to update its structure in this PR where not necessary :)

File: int/int_test.go (hunk 10 above)
Comment (NautiluX) [reply int/int_test.go]: copy-pasted comment, removed.

**Pull request merged**